src/vs/platform/terminal/common/terminalProcess.ts

```diff
@@ -0,0 +1,180 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { UriComponents } from 'vs/base/common/uri';
+import { IRawTerminalTabLayoutInfo, ITerminalEnvironment, ITerminalTabLayoutInfoById } from 'vs/platform/terminal/common/terminal';
+import { ISerializableEnvironmentVariableCollection } from 'vs/platform/terminal/common/environmentVariable';
+
+export interface IShellLaunchConfigDto {
+	name?: string;
+	executable?: string;
+	args?: string[] | string;
+	cwd?: string | UriComponents;
+	env?: { [key: string]: string | null; };
+	hideFromUser?: boolean;
+}
+
+export interface ISingleTerminalConfiguration<T> {
+	userValue: T | undefined;
+	value: T | undefined;
+	defaultValue: T | undefined;
+}
+
+export interface ICompleteTerminalConfiguration {
+	'terminal.integrated.automationShell.windows': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.automationShell.osx': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.automationShell.linux': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shell.windows': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shell.osx': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shell.linux': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shellArgs.windows': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shellArgs.osx': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.shellArgs.linux': ISingleTerminalConfiguration<string | string[]>;
+	'terminal.integrated.env.windows': ISingleTerminalConfiguration<ITerminalEnvironment>;
+	'terminal.integrated.env.osx': ISingleTerminalConfiguration<ITerminalEnvironment>;
+	'terminal.integrated.env.linux': ISingleTerminalConfiguration<ITerminalEnvironment>;
+	'terminal.integrated.inheritEnv': boolean;
+	'terminal.integrated.cwd': string;
+	'terminal.integrated.detectLocale': 'auto' | 'off' | 'on';
+	'terminal.flowControl': boolean;
+}
+
+export type ITerminalEnvironmentVariableCollections = [string, ISerializableEnvironmentVariableCollection][];
+
+export interface IWorkspaceFolderData {
+	uri: UriComponents;
+	name: string;
+	index: number;
+}
+
+export interface ICreateTerminalProcessArguments {
+	configuration: ICompleteTerminalConfiguration;
+	resolvedVariables: { [name: string]: string; };
+	envVariableCollections: ITerminalEnvironmentVariableCollections;
+	shellLaunchConfig: IShellLaunchConfigDto;
```

Comment:
I think a lot of this file isn't needed anymore after removing `events`/`_events`, if we do we should merge all this into common/terminal.ts since that's typically where we stick interfaces.

Reply:
we don't need it anymore so I removed it

Reply:
removed it